config/lambda/config.go

```diff
@@ -61,6 +62,14 @@ func Configure(p *config.Provider) {
 			SelectorFieldName: "SubnetIDSelector",
 		}
 		delete(r.TerraformResource.Schema, "filename")
+
+		r.TerraformCustomDiff = func(diff *terraform.InstanceDiff, _ *terraform.InstanceState, _ *terraform.ResourceConfig) (*terraform.InstanceDiff, error) {
+			if diff != nil && diff.Attributes != nil {
+				delete(diff.Attributes, "source_code_hash")
```

Comment:
From the [Terraform docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#source_code_hash), looks like this argument is used to trigger updates probably in situations where the image URI does not change but the contents of the image identified by that URI changes. 

So, we had better consider this argument while calculating the diffs. My understanding the problem here is Terraform expects from the users of this API to update the hash accordingly when the image URI (or more precisely when the content is) updated. If this is really the case, what we can do is to make this dependency explicit in the parameter's documentation.

In addition to this, we had better disable late-initialization for this field because if the above explanation is correct, then whenever a user specified an image URI but leaves the hash empty, could it be the case that the provider late-initializes this field with the computed hash, making it a requirement for the user to update the hash if they just want to update the image URI? If this is the case, then disabling late-initialization would relieve users of this API, who just rely on the image URI (and don't want to trigger an update by updating the hash).

Reply:
I think the usage of this is really for zip based lambda deployments (where contents are in S3). I don't think `source_code_hash` has anything to do with image based deployments. I have at least yet to see any documentation for container based Lambdas where this is referenced. I think maybe what we could do is see if the attribute for `image_uri` is not nil and if it is not nil then delete the `source_code_hash`?

Reply:
@ulucinar, after disabling late-initialization for `source_code_hash`, I do not encounter the update loop (unless the user has set `source_code_hash` manually) and I think it's better to continue this way.

@blakeromano, although I do not know the exact use case, after the tests I have done, I think that the diff for `source_code_hash` should not be deleted. In case the `image_uri` remains the same for an image in use and the `source_code_hash` changes, it is necessary to set the source_code_hash to update the resource.